v2/pkg/templates/cluster.go

```diff
@@ -78,6 +79,17 @@ func Cluster(list map[string]*Template) [][]*Template {
 				}
 			}
 		}
+		if len(template.RequestsSSL) == 1 {
+			for otherKey, other := range list {
+				if len(other.RequestsSSL) == 0 || len(other.RequestsSSL) > 1 {
+					continue
+				}
+				if template.RequestsSSL[0].CanCluster(other.RequestsSSL[0]) {
+					delete(list, otherKey)
+					cluster = append(cluster, other)
+				}
+			}
+		}
```

Comment:
@Ice3man543 , Thinking out load, This does have any impact . But I was thinking maybe we can check `templateType` when iterating over `list` . and if possible maybe we can use `switch` statements over nested if else

Reply:
Done, let me know if the changes look good!

Reply:
@Ice3man543 , lgtm!